benchmark/src/main/java/test/CacheReadBenchmark.java

```diff
@@ -0,0 +1,75 @@
+package test;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nonnull;
+
+import org.openjdk.jmh.annotations.*;
+
+import com.google.common.base.Optional;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import io.servicefabric.transport.utils.memoization.Computable;
+import io.servicefabric.transport.utils.memoization.ConcurrentMapMemoizer;
+
+@Fork(2)
+@State(Scope.Thread)
+@Threads(4)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class CacheReadBenchmark {
```

Comment:
> Here you can see memoizer x10 faster than guava_cache. In MessageSchema we can gain some win if switching it to memoizer.

Also, as I said earlier, you can move to JDK 8.
Then you can use [Map.computeIfAbsent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-) - it is also faster than Guava cache (but not so powerful).